### PR TITLE
fix: missing AutoencoderKL lora adapter

### PR DESCRIPTION
**Pull request by beniz**
# What does this PR do?

This PR fixes the missing lora adapter with VAE (AutoencoderKL class).
Discussion is here: https://github.com/huggingface/diffusers/discussions/9771
Related reports:
https://github.com/GaParmar/img2img-turbo/issues/64
https://github.com/radames/Real-Time-Latent-Consistency-Model/issues/38

## Who can review?

cc @sayakpaul

### CODE REVIEW

File: src/diffusers/models/autoencoders/autoencoder_kl.py [hunk 1]
```diff
@@ -31,10 +31,11 @@
 )
 from ..modeling_outputs import AutoencoderKLOutput
 from ..modeling_utils import ModelMixin
+from ...loaders import PeftAdapterMixin
 from .vae import Decoder, DecoderOutput, DiagonalGaussianDistribution, Encoder
 
 
-class AutoencoderKL(ModelMixin, ConfigMixin, FromOriginalModelMixin):
+class AutoencoderKL(ModelMixin, ConfigMixin, FromOriginalModelMixin, PeftAdapterMixin):
```
Comment (sayakpaul) [inline src/diffusers/models/autoencoders/autoencoder_kl.py:38]: Could we also add a test for this in https://github.com/huggingface/diffusers/blob/main/tests/models/autoencoders/test_models_vae.py?

**Review by sayakpaul — COMMENTED** [review]
Thank you! Just a single comment.

File: src/diffusers/models/autoencoders/autoencoder_kl.py (hunk 1 above)
Comment (beniz) [reply src/diffusers/models/autoencoders/autoencoder_kl.py:38]: Done, let me know if more/different is needed.

File: tests/models/autoencoders/test_models_vae.py [hunk 2]
```diff
@@ -846,6 +846,39 @@ def test_stable_diffusion_encode_sample(self, seed, expected_slice):
         assert torch_all_close(output_slice, expected_output_slice, atol=tolerance)
 
 
+    def test_lora_adapter(self):
+        
+        vae = AutoencoderKL.from_pretrained("stabilityai/sd-turbo", subfolder="vae")
```
Comment (sayakpaul) [inline tests/models/autoencoders/test_models_vae.py:851]: Ah it's a bit large to test in a PR. Can we leverage something like so?

```python
init_dict, inputs_dict = self.prepare_init_args_and_inputs_for_common()
model = self.model_class(**init_dict)

vae_lora_config = LoraConfig(
            r=16,
            init_lora_weights="gaussian",
            target_modules=target_modules_vae,
        )

vae.add_adapter(vae_lora_config, adapter_name="vae_lora")

active_lora = vae.active_adapters()
self.assertTrue(len(active_lora) == 1)
self.assertTrue(active_lora[0] == "vae_lora")
```

Of course, we would need to update the `target_modules_vae` accordingly as well.

LMK if it's unclear.

Comment (beniz) [reply tests/models/autoencoders/test_models_vae.py]: Thanks @sayakpaul for the code snippet, this helped me figure that I had added the test to the wrong class. It does successfully execute and pass the diffusers tests on my side.

**Comment by yiyixuxu** [comment]
can you run `make style`?

**Comment by yiyixuxu** [comment]
the failing tests are related, can we look into them?

I think we may need to add a @require_peft_backend similar to https://github.com/huggingface/diffusers/blob/5588725e8e7be497839432e5328c596169385f16/tests/models/unets/test_models_unet_2d_condition.py#L1081

**Review by sayakpaul — COMMENTED** [review]
Sorry I had actually forgot to submit my reviews.

**Comment by beniz** [comment]
> the failing tests are related, can we look into them?
> 
> I think we may need to add a @require_peft_backend similar to
> 
> https://github.com/huggingface/diffusers/blob/5588725e8e7be497839432e5328c596169385f16/tests/models/unets/test_models_unet_2d_condition.py#L1081

All done @yiyixuxu I believe, let me know if anything else remains.

File: tests/models/autoencoders/test_models_vae.py [hunk 3]
```diff
@@ -299,7 +299,38 @@ def test_output_pretrained(self):
 
         self.assertTrue(torch_all_close(output_slice, expected_output_slice, rtol=1e-2))
 
+    def test_lora_adapter(self):
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply tests/models/autoencoders/test_models_vae.py:267]: @beniz seems like this was not resolved?

[the remark this answers is not in the document]
Comment (beniz) [reply tests/models/autoencoders/test_models_vae.py:307]: @sayakpaul ah apologies, may have forgot to push to repo. Done. Thanks for your vigilance.

**Comment by sayakpaul** [comment]
@beniz thanks! Possible to fix the quality issues by running `make style && make quality`?

**Comment by beniz** [comment]
> @beniz thanks! Possible to fix the quality issues by running `make style && make quality`?

I've fix a missing dependency. FYI running make style && qualilty fails on other unrelated files, so I've been looking at the underlying ruff calls to apply them to the relevant files.

**Comment by sayakpaul** [comment]
@beniz I pushed the quality fixes directly to your branch, which I hope is okay. If not, please let me know, I will revert immediately

**Comment by beniz** [comment]
> @beniz I pushed the quality fixes directly to your branch, which I hope is okay. If not, please let me know, I will revert immediately

Much appreciated, thank you.

**Review by sayakpaul — APPROVED** [review]

**Pull request merged**